backend/src/main/java/gov/cdc/usds/simplereport/service/crm/DynamicsCrmProvider.java

```diff
@@ -68,6 +68,11 @@ private String getAccessTokenFromService() throws InterruptedException {
               null);
 
       credentials = future.get();
+      if (credentials == null) {
+        LOG.error("Dynamics AuthenticationResult is null");
```

Comment:
Should we alert pager duty when this is the case? This seem to imply that our Dynamics integration is broken

Reply:
I think this was a false flag from Coverity and the condition cannot actually be met since the `ExecuterService.submit()` is passed a `Callable` (of class `AcquireTokenCallable`).  I don't understand it fully enough to say it is impossible, but I believe `.get()` should only be able to be null if the service is executing a `Runnable`.

That said, yes, we need some visibility into what is happening in async tasks.  I was looking into using Application Insights `TelemetryClient` to track events so we can start seeing some of the happenings in here.  I'll add a ticket for that.